qp/pdf.py

```diff
@@ -60,14 +60,46 @@ def quantize(self, percent=1., number=None):
         print("Result: ", self.quantiles)
         return self.quantiles
 
-    def interpolate(self, number=100):
+    def interpolate(self):#, number=100, grid=None):
+        """
+        Returns interpolator based on quantiles.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.interpolator: interpolator
+            The interpolating function
+        """
+#         if self.interpolator is None:
+
+        if self.quantiles is None:
+            self.quantiles = self.quantize()
+        self.difs = self.quantiles[1:]-self.quantiles[:-1]
+        self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
+        self.quantvals = (1.0/(len(self.quantiles)+1))/self.difs
+
+        print("Creating interpolator")
+        self.interpolator = spi.interp1d(self.mids, self.quantvals, fill_value="extrapolate")
+
+#         if grid is None:
+#             grid = np.linspace(min(self.mids), max(self.mids), number)
+
+        return self.interpolator
+
+    def approximate(self, points):
+
         """
         Interpolates the quantiles.
 
         Parameters
         ----------
         number: int
             The number of points over which to interpolate, bounded by the quantile value endpoints
+        grid: ndarray
```

Comment:
What happens if you pass in a single floating point value for `points`? Approximate should return the value of the approximate PDF at that point, as a float, I think... BTW in the docstring you should have "points" instead of "grid"